### ISSUE TITLE
Blocks page and uses high memory footprint right after every page load on Firefox

### ISSUE THREAD

**Issue by mowgli** [uBlockOrigin/uMatrix-issues#118]
### Prerequisites

- [x] I performed a cursory search of the issue tracker to avoid opening a duplicate issue
    - Your issue may already be reported.
    - [x] I also searched the existing issues at <https://github.com/gorhill/uMatrix/issues>
- [x] This is not a support issue or a question
    - Support issues and questions are handled at [/r/uMatrix](https://old.reddit.com/r/uMatrix/)
- I tried to reproduce the issue when...
    - [ ] uMatrix is the only extension (not possible as then I will have no network connection)
    - [x] uMatrix with default lists/settings
    - [ ] using a new, unmodified browser profile (not possible as then I will have no network connection)
- [x] I am running the latest version of uMatrix
- [x] I checked the [documentation](https://github.com/gorhill/uMatrix/wiki) to understand that the issue I report is not a normal behavior
- [ ] I used the logger to rule out that the issue is caused by my ruleset

### Description

Since many time I suffer by a really bad firefox performance right after page load, Klicks are not used and my memory display shows a raise and a fall after 2-3 seconds. Interactive stuff is nearly not usable.

Today I found the time to debug the issue by disabling one addon after another until I found the problematic one. After I disabled umatrix, the pages working as I expected them to do.

So first I tried to set all umatrix settings to default but that didn't help.

Parallel to umatrix I also use ublock origin and use the blacklists there instead of umatrix.

I have no idea why it has such a behaviour. As I understand umatrix, it should work _bevore_ the page is completed, not afterwards.

### Steps to Reproduce

1. Load a page with umatrix enabled
2. Watch memory usage or try to interact with the page in the first few seconds

### Ruleset

{
  "app": "uMatrix",
  "version": "1.3.16",
  "when": 1551386969250,
  "settings": {
    "alwaysDetachLogger": false,
    "autoUpdate": false,
    "clearBrowserCache": false,
    "clearBrowserCacheAfter": 60,
    "cloudStorageEnabled": false,
    "collapseBlacklisted": true,
    "collapseBlocked": false,
    "colorBlindFriendly": false,
    "deleteCookies": true,
    "deleteUnusedSessionCookies": false,
    "deleteUnusedSessionCookiesAfter": 60,
    "deleteLocalStorage": true,
    "displayTextSize": "13px",
    "externalHostsFiles": [],
    "externalRecipeFiles": [],
    "iconBadgeEnabled": true,
    "maxLoggedRequests": 1000,
    "noTooltips": false,
    "popupCollapseAllDomains": false,
    "popupCollapseBlacklistedDomains": false,
    "popupScopeLevel": "domain",
    "processHyperlinkAuditing": true,
    "processReferer": false,
    "selectedHostsFiles": [],
    "selectedRecipeFiles": [],
    "userHosts": {
      "enabled": false,
      "content": ""
    },
    "userRecipes": {
      "enabled": false,
      "content": ""
    }
  },
  "rules": [
    "* * * block",
    "* * cookie block",
    "* * css allow",
    "* * frame block",
    "* * image allow",
    "* 1st-party * allow",
    "* 1st-party frame allow",
    "* noscript-csp.invalid * block",
    "autorouter.aero 1st-party cookie allow",
    "aviationexam.com ajax.googleapis.com script allow",
    "aviationexam.com aviationexam.com cookie allow",
    "cacert.org 1st-party cookie allow",
    "cpan.org pimg.net * allow",
    "dfs.de dfs.de cookie allow",
    "dilbert.com assets.amuniversal.com image allow",
    "fgzo.ch 1st-party cookie allow",
    "flightnet.aero fgzo.ch image allow",
    "flightnet.aero flightnet.aero cookie allow",
    "gentoo.org 1st-party cookie allow",
    "github.com cloud.githubusercontent.com * allow",
    "github.com github-production-release-asset-2e65be.s3.amazonaws.com xhr allow",
    "github.com github.com cookie allow",
    "github.com githubusercontent.com image allow",
    "github.com raw.githubusercontent.com * allow",
    "github.com render.githubusercontent.com * allow",
    "golem.de vgwort.de * block",
    "google.ch gstatic.com image allow",
    "google.com * script block",
    "google.de gstatic.com image allow",
    "heise.de gzhls.at * block",
    "heise.de ix.de * allow",
    "heise.de mpnrs.com * block",
    "https-strict: * true",
    "https-strict: avherald.com false",
    "https-strict: behind-the-scene false",
    "https-strict: flightnet.aero false",
    "matrix-off: about-scheme true",
    "matrix-off: behind-the-scene true",
    "matrix-off: chrome-extension-scheme true",
    "matrix-off: chrome-scheme true",
    "matrix-off: localhost true",
    "matrix-off: moz-extension-scheme true",
    "matrix-off: opera-scheme true",
    "mozilla.org mozilla.net * allow",
    "no-workers: * true",
    "no-workers: phzh.ch false",
    "noscript-spoof: twitter.com true",
    "onion * media block",
    "onion * script block",
    "reddit.com redditmedia.com css allow",
    "reddit.com redditmedia.com image allow",
    "reddit.com redditstatic.com css allow",
    "reddit.com redditstatic.com image allow",
    "referrer-spoof: * true",
    "referrer-spoof: behind-the-scene false",
    "serverfault.com sstatic.net css allow",
    "serverfault.com sstatic.net image allow",
    "skydemon.aero skydemon.aero cookie allow",
    "slideshare.net slidesharecdn.com css allow",
    "slideshare.net slidesharecdn.com image allow",
    "stackexchange.com sstatic.net * allow",
    "stackoverflow.com sstatic.net * allow",
    "sueddeutsche.de sueddeutsche.com image allow",
    "twitter.com * script block",
    "wikibooks.org wikimedia.org * allow",
    "wikipedia.org wikimedia.org * allow",
    "www.heise.de cloudimg.io image inherit",
    "xing.com 1st-party cookie allow",
    "youtube.com gstatic.com image allow",
    "youtube.com youtube-nocookie.com image allow",
    "youtube.com ytimg.com css allow",
    "youtube.com ytimg.com image allow"
  ],
  "rawSettings": {
    "contributorMode": false,
    "disableCSPReportInjection": false,
    "enforceEscapedFragment": true,
    "placeholderBackground": "url(\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAoAAAAKCAAAAACoWZBhAAAABGdBTUEAALGPC/xhBQAAAAJiS0dEAP+Hj8y/AAAAB3RJTUUH3wwIAAgyL/YaPAAAACJJREFUCFtjfMbOAAQ/gZiFnQPEBAEmGIMIJgtIL8QEgtoAIn4D/96X1KAAAAAldEVYdGRhdGU6Y3JlYXRlADIwMTUtMTItMDhUMDA6MDg6NTArMDM6MDAasuuJAAAAJXRFWHRkYXRlOm1vZGlmeQAyMDE1LTEyLTA4VDAwOjA4OjUwKzAzOjAwa+9TNQAAAABJRU5ErkJggg==\") repeat scroll #fff",
    "placeholderBorder": "1px solid rgba(0, 0, 0, 0.1)",
    "imagePlaceholder": true,
    "imagePlaceholderBackground": "default",
    "imagePlaceholderBorder": "default",
    "framePlaceholder": true,
    "framePlaceholderDocument": "<html><head><meta charset=\"utf-8\"><style>body { background: {{bg}};color: gray;font: 12px sans-serif;margin: 0;overflow: hidden;padding: 2px;white-space: nowrap;}a { color: inherit;padding: 0 3px;text-decoration: none;}svg {display: inline-block;fill: gray;height: 12px;vertical-align: bottom;width: 12px;}</style></head><body><span><a href=\"{{url}}\" title=\"{{url}}\" target=\"_blank\"><svg viewBox=\"0 0 1792 1792\"><path transform=\"scale(1,-1) translate(0,-1536)\" d=\"M1408 608v-320q0 -119 -84.5 -203.5t-203.5 -84.5h-832q-119 0 -203.5 84.5t-84.5 203.5v832q0 119 84.5 203.5t203.5 84.5h704q14 0 23 -9t9 -23v-64q0 -14 -9 -23t-23 -9h-704q-66 0 -113 -47t-47 -113v-832q0 -66 47 -113t113 -47h832q66 0 113 47t47 113v320q0 14 9 23t23 9h64q14 0 23 -9t9 -23zM1792 1472v-512q0 -26 -19 -45t-45 -19t-45 19l-176 176l-652 -652q-10 -10 -23 -10t-23 10l-114 114q-10 10 -10 23t10 23l652 652l-176 176q-19 19 -19 45t19 45t45 19h512q26 0 45 -19t19 -45z\" /></svg></a>{{url}}</span></body></html>",
    "framePlaceholderBackground": "default"
  }
}

### Supporting evidence

I would like to but how to be fast enough to klick on the page and do the screen shoot right afterwards?

### Your environment

* uMatrix version: 1.3.16
* Browser Name and version: Firefox 65.0.1 64bit
* Operating System and version: Linux Debian derivate or Gentoo

**Comment by mowgli** [comment]
Forget about. Now I have the issue even with that addon completly disabled. :-(

**Comment by uBlock-user** [comment]
> Speculated performance issues will be marked as invalid and closed if they do not come with actual profiling data + analysis supporting the claim.
